lib/note-detail.jsx

```diff
@@ -60,6 +68,21 @@ export default React.createClass( {
 		}
 	},
 
+	onEditorChanged: function() {
+		clearTimeout( saveTimerId );
+		// Save the note after user stops typing
+		saveTimerId = delay( this.saveNote, saveDelay );
+	},
```

Comment:
I believe you are looking for `debounce()` which would let you remove the timer handles and delay variables.

```js
onEditorChanged: debounced( this.saveNote, saveDelay )
```

What `debounce()` does is wait until the delay period has finished before actually running the function. If more calls to this function come in during that waiting period, it resets the clock. Additionally, it has a few options which can still help us out, such as the `maxWait` property. `maxWait` will guarantee that the function gets called at least as often as the number of milliseconds given in. That is, we could tell the app to wait to save the note until the user stops typing for three seconds, but if they continue to type, we could save it at least every twenty seconds.

```js
import { debounce } from 'lodash';
onEditorChanged: debounced( this.saveNote, saveDelay, { maxWait: 30000 } );
```